includes/payment-methods/class-upe-split-payment-gateway.php

```diff
@@ -119,8 +119,8 @@ public function register_scripts() {
 		if ( $this->supports( 'tokenization' ) ) {
 			$script_dependencies[] = 'woocommerce-tokenization-form';
 		}
-		WC_Payments::register_script_with_dependencies( 'wcpay-upe-checkout', 'dist/upe_split_checkout', $script_dependencies );
-
+		$script = ( WC_Payments_Features::is_upe_deferred_intent_enabled() && ! WC_Payments_Features::is_upe_split_enabled() ) ? 'dist/upe_with_deferred_intent_creation_checkout' : 'dist/upe_split_checkout';
```

Comment:
```suggestion
		$script = ( WC_Payments_Features::is_upe_deferred_intent_enabled() ) ? 'dist/upe_with_deferred_intent_creation_checkout' : 'dist/upe_split_checkout';
```
Presumably if we reach this point, then at least one of `is_upe_deferred_intent_enabled()` and `is_upe_split_enabled()` is `true`. The logic here ensures that if the latter is true--and the original split-UPE flag is enabled--the new deferred intent flag would be overridden and the original split-UPE would be returned instead. Is this intended/expected?

As with the split-UPE release--where when the split-UPE is eligible and enabled, all other UPE flags are ignored--it makes sense to me to have the deferred intent flag similarly override all of its ancestors as well. That way we can simply enable the feature by enabling the single flag, without having to worry about disabling the others. 

Let me know your thoughts or if I'm missing anything here!

Reply:
The current implementation is something I introduced consciously, and I think I brought it one day to our catch-up. I wanted to bring an extra layer of protection for the deferred UPE and ignore it when it's enabled with the split UPE. I thought we'd review the enablement strategy as part of our current dedicated issue in our backlog. That's the background.

I have no objections and can definitely make the deferred intent UPE a priority across all the UPEs. I updated the code in 4de930d1f392ffaa0b1df32201ae7a482c26ba5a. Please not, that I also changed the order between the legacy UPE and the deferred intent UPE in one place, in order to make deferred UPE work as expected, when both flags are enabled. 

Having that said, I would vote for testing out all the 7 possible combinations of feature flags that are currently possible, then noting down those that for some reason should be amended and consider addressing them as part of the enablement strategy issue or any other issue on an earlier stage. Here are the possible combinations: 

Example: 
| Legacy UPE | Split UPE | Deferred UPE | The rolled out UPE     |
|------------|-----------|--------------|-------------|
| Enabled    | Enabled   | Enabled      | deferred |
| Enabled    | Enabled   | Disabled     | split |
| Enabled    | Disabled  | Enabled      | deferred |
| Enabled    | Disabled  | Disabled     | legacy |
| Disabled   | Enabled   | Enabled      | deferred |
| Disabled   | Enabled   | Disabled     | split |
| Disabled   | Disabled  | Enabled      | deferred |
| Disabled   | Disabled  | Disabled     | legacy card |